app/controllers/api/v0/service_offering_icons_controller.rb

```diff
@@ -4,6 +4,15 @@ class ServiceOfferingIconsController < ApplicationController
       include Api::V0::Mixins::IndexMixin
       include Api::V0::Mixins::ShowMixin
 
+      def icon_data
+        service_offering_icon = model.find(params[:service_offering_icon_id].to_i)
+
+        send_data(service_offering_icon.data, 
+          :type         => MimeMagic.by_magic(service_offering_icon.data).type,
```

Comment:
@agrare @bdunne are we ok with new gem for recognizing the content type?

Reply:
Seems ok to me.